Makefile

```diff
@@ -12,18 +12,179 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+MLRUN_DOCKER_TAG := $(if $(MLRUN_DOCKER_TAG),$(MLRUN_DOCKER_TAG),latest)
+MLRUN_DOCKER_REPO := $(if $(MLRUN_DOCKER_REPO),$(MLRUN_DOCKER_REPO),mlrun)
+MLRUN_DOCKER_REGISTRY := $(if $(MLRUN_DOCKER_REGISTRY),$(MLRUN_DOCKER_REGISTRY),) #e.g. quay.io/
+MLRUN_ML_DOCKER_IMAGE_NAME_PREFIX := $(if $(MLRUN_ML_DOCKER_IMAGE_NAME_PREFIX),$(MLRUN_ML_DOCKER_IMAGE_NAME_PREFIX),ml-)
+MLRUN_PACKAGE_TAG := $(if $(MLRUN_PACKAGE_TAG),$(MLRUN_PACKAGE_TAG),development)
+MLRUN_GITHUB_REPO := $(if $(MLRUN_GITHUB_REPO),$(MLRUN_GITHUB_REPO),mlrun)
+MLRUN_ML_PYTHON_VERSION := $(if $(MLRUN_ML_PYTHON_VERSION),$(MLRUN_ML_PYTHON_VERSION),3.8)
+MLRUN_LEGACY_ML_PYTHON_VERSION := $(if $(MLRUN_LEGACY_ML_PYTHON_VERSION),$(MLRUN_LEGACY_ML_PYTHON_VERSION),3.6)
+MLRUN_API_PYTHON_VERSION := $(if $(MLRUN_API_PYTHON_VERSION),$(MLRUN_API_PYTHON_VERSION),3.7)
```

Comment:
perhaps use the same version across ML & API?

Reply:
Needed @yjb-ds approval for this - approved, we will have:
`MLRUN_PYTHON_VERSION` and `MLRUN_LEGACY_ML_PYTHON_VERSION`